webpack/assets/javascripts/react_app/components/common/forms/FormField.js

```diff
@@ -0,0 +1,140 @@
+import { omit } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Col,
+  FormGroup,
+  FormControl,
+  ControlLabel,
+  HelpBlock,
+  FieldLevelHelp,
+} from 'patternfly-react';
+
+import { noop } from '../../../common/helpers';
+import DateTimePicker from '../DateTimePicker/DateTimePicker';
+import DatePicker from '../DateTimePicker/DatePicker';
+import TimePicker from '../DateTimePicker/TimePicker';
+
+const inputComponents = {
+  date: DatePicker,
+  dateTime: DateTimePicker,
+  time: TimePicker,
+};
+const componentTypes = Object.keys(inputComponents);
+const inputTypes = componentTypes.concat(['text']);
+
+const FormField = ({
+  type,
+  id,
+  name,
+  className,
+  disabled,
+  required,
+  touched,
+  error,
+  value,
+  label,
+  labelHelp,
+  labelSizeClass,
+  inputSizeClass,
+  onChange,
+  ...props
+}) => {
+  const validationState = touched && error ? 'error' : null;
+
+  return (
+    <FormGroup
+      controlId={id}
+      disabled={disabled}
+      validationState={validationState}
+    >
+      <ControlLabel className={labelSizeClass}>
+        {label}
+        {labelHelp && (
+          <FieldLevelHelp
+            placement="right"
+            buttonClass="field-help"
+            content={<React.Fragment>{labelHelp}</React.Fragment>}
+          />
+        )}
+        {required && '*'}
```

Comment:
nit: you may need to swap `labelHelp` and `required`: https://github.com/theforeman/foreman/pull/6495

Reply:
:+1: thanks